packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -0,0 +1,367 @@
+import { fetchI18nBundle, getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { isPhone } from "@ui5/webcomponents-base/dist/Device.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import Dialog from "@ui5/webcomponents/Dialog.js";
+import Button from "@ui5/webcomponents/Button.js";
+import Bar from "./Bar.js";
+import GroupHeaderListItem from "@ui5/webcomponents/dist/GroupHeaderListItem.js";
+import List from "@ui5/webcomponents/dist/List.js";
+import StandardListItem from "@ui5/webcomponents/dist/StandardListItem.js";
+
+import {
+	VSD_DIALOG_TITLE_SORT,
+	VSD_SUBMIT_BUTTON,
+	VSD_CANCEL_BUTTON,
+	VSD_RESET_BUTTON,
+	VSD_SORT_ORDER,
+	VSD_SORT_BY,
+	VSD_ORDER_ASCENDING,
+	VSD_ORDER_DESCENDING,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Template
+import ViewSettingsDialogTemplate from "./generated/templates/ViewSettingsDialogTemplate.lit.js";
+
+// Styles
+import viewSettingsDialogCSS from "./generated/themes/ViewSettingsDialog.css.js";
+
+const metadata = {
+	tag: "ui5-view-settings-dialog",
+	managedSlots: true,
+	properties: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the initial sort order.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 sortDescending: {
+			type: Boolean,
+		},
+		/**
+		 * Keeps recently focused list in order to focus it on next dialog open.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _recentlyFocused: {
+			type: Object,
+
+		},
+		/**
+		 * Stores settings of the dialog before the first open.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _initialSettings: {
+			type: Object,
+		},
+		/**
+		 * Stores settings of the dialog after confirmation.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _confirmedSettings: {
+			type: Object,
+		},
+		/**
+		 * Stores current settings of the dialog.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _currentSettings: {
+			type: Object,
+		},
+	},
+	slots: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the <code>sortItems</code> list.
+		 * @type {sap.ui.webcomponents.fiori.ListItem}
+		 * @slot sortItems
+		 * @public
+		 */
+		 "sortItems": {
+			propertyName: "sortItems",
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+
+		/**
+		 * Fired when OK button is activated.
```

Comment:
"OK" button is not completely accurate, in future this might come configurable and become "Sort", "Confirm", etc. Perhaps confirmation button is better (the event is also called confirm)

Reply:
Done